### PR TITLE
relay JS exceptions to Sentry via Raven

### PR DESCRIPTION
**Pull request by ericcf**
* capture JS exceptions and as much metadata as possible, POST
  to the app and then relay to Sentry

[##92255192]

### CODE REVIEW

File: app/assets/javascripts/capture_exceptions.js.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+(function() {
+  'use strict';
+
+  function trackException(data) {
+    window.$.post('<%= url.exception_report_path %>', data);
+  }
+
+  window.onerror = function(message, url, line, col, error){
+    var stopPropagation = window.debug ? false : true;
```
Comment (michaelwehrley) [inline app/assets/javascripts/capture_exceptions.js.erb]: When/where does `window.debug` get set?

Comment (ericcf) [inline app/assets/javascripts/capture_exceptions.js.erb]: Added on line 2.

File: spec/controllers/exception_reports_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+
+RSpec.describe ExceptionReportsController, type: :controller do
+  describe "POST create" do
+    context "when unauthenticated" do
+      it "does nothing" do
+        allow(controller).to receive(:current_user)
```
Comment (michaelwehrley) [inline spec/controllers/exception_reports_controller_spec.rb]: I never thought about not returning anything would still be_falsey.  :golf:

**Comment by michaelwehrley** [comment]
:eyes:  good!

**Pull request merged**